### PR TITLE
Fix Integer Rounding issue and duplicate definition of __deref in sal.h

### PR DESCRIPTION
**Pull request by franksinankaya**

### CODE REVIEW

**Comment by franksinankaya** [comment]
@jkotas @am11

File: src/pal/inc/rt/sal.h [hunk 1]
```diff
@@ -2604,6 +2604,7 @@ extern "C" {
 #ifndef PAL_STDCPP_COMPAT
     #define __null
     #define __notnull
+    #define __deref
```
Comment (franksinankaya) [inline src/pal/inc/rt/sal.h]: I'm not sure if I should have done it for GCC only or not. I guess we'll see the build results.

**Comment by franksinankaya** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test 
@dotnet-bot test coreclr-ci 
@dotnet-bot test coreclr-ci (Test Pri0 Windows_NT x64 checked)  
@dotnet-bot test coreclr-ci (Test Pri0 Windows_NT x86 checked)

**Comment by franksinankaya** [comment]
@dotnet-bot test coreclr-ci
@dotnet-bot test coreclr-ci (Test Pri0 Windows_NT x64 checked)
@dotnet-bot test coreclr-ci (Test Pri0 Windows_NT x86 checked)

**Comment by franksinankaya** [comment]
Rebased to see if these failures will go away.

**Comment by am11** [comment]
Current master's src/pal is building with GCC on Ubuntu 18.04. Are you having different experience?

**Comment by franksinankaya** [comment]
There were two coreclr-ci failures previously. I'm trying to figure out if those failed because of this change or something else.

No issues with GCC compilation.

**Comment by franksinankaya** [comment]
@dotnet-bot test coreclr-ci
@dotnet-bot test coreclr-ci (Test Pri0 Linux_musl arm64 checked)
@dotnet-bot test coreclr-ci (Test pri0 Linux arm64 checked)

**Pull request merged**